pkg/sdn/plugin/node.go

```diff
@@ -125,18 +149,140 @@ func (node *OsdnNode) Start() error {
 		}
 		for _, p := range pods {
 			containerID := getPodContainerID(&p)
-			err = node.UpdatePod(p.Namespace, p.Name, kubeletTypes.DockerID(containerID))
+			err = node.UpdatePod(p.Namespace, p.Name, kubeletTypes.ContainerID{ID: containerID})
 			if err != nil {
 				log.Warningf("Could not update pod %q (%s): %s", p.Name, containerID, err)
 			}
 		}
 	}
 
+	if err := node.cniServer.Start(); err != nil {
+		return err
+	}
+	go utilwait.Forever(node.watchCni, 0)
+
 	node.markPodNetworkReady()
 
 	return nil
 }
 
+type podWorker struct {
+	node     *OsdnNode
+	requests chan *podRequest
+}
+
+func newPodWorker(node *OsdnNode) *podWorker {
+	return &podWorker{
+		node: node,
+		// Allow queue depth of 10 before blocking additions, if there
+		// are more than 10 outstanding requests something is wrong
+		requests: make(chan *podRequest, 10),
+	}
+}
+
+func (w *podWorker) addRequest(request *podRequest) {
+	w.requests <- request
+	if len(w.requests) == 1 {
```

Comment:
What are you trying to do here?  If you're trying to be clever and only run a goroutine while requests are pending that's not really going to make this more efficient.  Also, is this called within a lock / single threaded context?  If not this code is racy.